src/App.jsx

```diff
@@ -14,33 +16,102 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const goods = goodsFromServer.map((name, id) => ({
+  name,
+  id,
+}));
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+const SORT_FIELD_ALPHABETIC = 'Alphabetically';
+const SORT_FIELD_BYLENGTH = 'ByLength';
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+function getPreparedGood(allGoods, { sortField, isReversed }) {
+  const visibleGoods = [...allGoods];
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (sortField) {
+    visibleGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_ALPHABETIC:
+          return good1.name.localeCompare(good2.name);
+
+        case SORT_FIELD_BYLENGTH:
+          return good1.name.length - good2.name.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getPreparedGood(goods, { sortField, isReversed });
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const handleReset = () => {
+    setSortField('');
+    setIsReversed(false);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortField !== SORT_FIELD_ALPHABETIC,
+          })}
+          onClick={() => setSortField(SORT_FIELD_ALPHABETIC)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortField !== SORT_FIELD_BYLENGTH,
+          })}
+          onClick={() => setSortField(SORT_FIELD_BYLENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={() => setIsReversed(!isReversed)}
+        >
+          Reverse
+        </button>
+
+        {sortField || isReversed ? (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={handleReset}
+          >
+            Reset
+          </button>
+        ) : (
+          ''
+        )}
```

Comment:
```suggestion
        {sortField || isReversed && (
          <button
            type="button"
            className="button is-danger is-light"
            onClick={handleReset}
          >
            Reset
          </button>
        )
```
You can simplify, or return null